File: src/Raven.Server/Dashboard/IndexingSpeed.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Sparrow.Json.Parsing;

namespace Raven.Server.Dashboard
{
    public class IndexingSpeed : AbstractDashboardNotification
    {
        public List<IndexingSpeedItem> Items { get; set; }

        public IndexingSpeed()
        {
            Items = new List<IndexingSpeedItem>();
        }
    }

    public class IndexingSpeedItem : IDynamicJson
    {
        public string Database { get; set; }

        public double IndexedPerSecond { get; set; }

        public double MappedPerSecond { get; set; }

        public double ReducedPerSecond { get; set; }

        public DynamicJsonValue ToJson()
        {
            return new DynamicJsonValue
            {
                [nameof(Database)] = Database,
                [nameof(IndexedPerSecond)] = IndexedPerSecond,
                [nameof(MappedPerSecond)] = MappedPerSecond,
                [nameof(ReducedPerSecond)] = ReducedPerSecond
            };
        }
    }
}

File: src/Raven.Studio/typescript/models/resources/serverDashboard/indexingSpeed.ts
/// <reference path="../../../../typings/tsd.d.ts"/>

class indexingSpeed {
    database = ko.observable<string>();
    
    indexedPerSecond = ko.observable<number>();
    mappedPerSecond = ko.observable<number>();
    reducedPerSecond = ko.observable<number>();
    
    showOnGraph = ko.observable<boolean>();
    
    constructor(dto: Raven.Server.Dashboard.IndexingSpeedItem) {
        this.update(dto);
    }
    
    update(dto: Raven.Server.Dashboard.IndexingSpeedItem) {
        this.database(dto.Database);
        this.indexedPerSecond(this.round(dto.IndexedPerSecond));
        this.mappedPerSecond(this.round(dto.MappedPerSecond));
        this.reducedPerSecond(this.round(dto.ReducedPerSecond));
    }

    private round(num: number): number {
        if (num < 1) {
            return num;
        }

        return Math.round(num);
    }
}

export = indexingSpeed;
